src/lib/utils/letterDataUtils.test.ts

```diff
@@ -93,3 +101,173 @@ describe('makeWordUsed', () => {
 		});
 	});
 });
+
+describe('provideHint', () => {
+	it('should populate a hint into playedLetters', () => {
+		let playedLetters: LetterData[] = [
+			{ value: 'h', state: 'firstLetter' },
+			{ value: 'e', state: 'hint' },
+			{ value: 'o', state: 'inPlay' }
+		];
+
+		let unplayedLetters: LetterData[] = [
+			{ value: 'l', state: 'inPlay' },
+			{ value: 'l', state: 'inPlay' }
+		];
+
+		[playedLetters, unplayedLetters] = provideHint(playedLetters, unplayedLetters, 'hello');
+
+		const expectedPlayedLetters: LetterData[] = [
+			{ value: 'h', state: 'firstLetter' },
+			{ value: 'e', state: 'hint' },
+			{ value: 'l', state: 'hint' }
+		];
+
+		const expectedUnplayedLetters: LetterData[] = [
+			{ value: 'l', state: 'inPlay' },
+			{ value: 'o', state: 'inPlay' }
+		];
+
+		expect(playedLetters).toEqual(expectedPlayedLetters);
+		expectArrayEquivalence(unplayedLetters, expectedUnplayedLetters);
+	});
+	describe('given a previous word has been solved', () => {
```

Comment:
nit: you should give your blocks some breathing room
```suggestion
	});
	
	describe('given a previous word has been solved', () => {
```